cdap-ui/app/directives/timeline-preview/timeline-ctrl-preview.js

```diff
@@ -0,0 +1,188 @@
+      }
+    });
+  };
+
+  $scope.$on(caskWindowManager.event.resize, () => {
+    $timeout($scope.initialize);
+  });
+  this.updateTotalLogsInStore = function(val) {
+    LogViewerStore.dispatch({
+      type: LOGVIEWERSTORE_ACTIONS.TOTAL_LOGS,
+      payload: {
+        totalLogs: val
+      }
+    });
+  };
+  this.updateScrollPositionInStore = function(val) {
+    LogViewerStore.dispatch({
+      type: LOGVIEWERSTORE_ACTIONS.SCROLL_POSITION,
+      payload: {
+        scrollPosition: val
+      }
+    });
+  };
+
+  this.updateTotalErrorsInStore = function(val) {
+    LogViewerStore.dispatch({
+      type: LOGVIEWERSTORE_ACTIONS.TOTAL_ERRORS,
+      payload: {
+        totalErrors: val
+      }
+    });
+  };
+
+  this.updateTotalWarningsInStore = function(val) {
+    LogViewerStore.dispatch({
+      type: LOGVIEWERSTORE_ACTIONS.TOTAL_WARNINGS,
+      payload: {
+        totalWarnings: val
+      }
+    });
+  };
+
+  var pollPromise = null;
+  var apiSettings = {
+    metric : {
+      context:  `namespace.${this.namespaceId}.app.${this.previewId}`,
+      names: ['system.app.log.error', 'system.app.log.warn', 'system.app.log.info', 'system.app.log.debug'],
+      startTime : '',
+      endTime : '',
+      resolution: 'auto'
+    }
+  };
+
+  this.setDefaultTimeWindow = () => {
+    apiSettings.metric.startTime = '';
+    apiSettings.metric.endTime = '';
+    this.updateStartTimeInStore(apiSettings.metric.startTime);
+  };
+
+  const getStatus = () => {
+    if (!pollPromise) {
+      return;
+    }
+    myPreviewLogsApi.getLogsStatus({
+      namespace : this.namespaceId,
+      previewId : this.previewId
+    }).$promise.then(
+      (res) => {
+        let previewStatus =  ['KILLED', 'COMPLETED', 'FAILED', 'STOPPED', 'KILLED_BY_TIMER'];
+        if (pollPromise && previewStatus.indexOf(res.status) !== -1) {
+          dataSrc.stopPoll(pollPromise.__pollId__);
+          pollPromise = null;
+        }
+      },
+      (err) => {
+        if (pollPromise) {
+          dataSrc.stopPoll(pollPromise.__pollId__);
+          pollPromise = null;
+        }
+        console.log('ERROR: ', err);
+      });
+  };
+
+  const pollForMetadata = () => {
+    pollPromise = dataSrc.poll({
+      _cdapPath: `/namespaces/${this.namespaceId}/previews/${this.previewId}/metrics/query`,
+      method: 'POST',
+      body: MyMetricsQueryHelper.constructQuery(
+        'qid',
+        MyMetricsQueryHelper.contextToTags(apiSettings.metric.context),
+        apiSettings.metric
+      )
+    },
+    (res) => {
+      $scope.metadata = res;
+      $scope.sliderBarPositionRefresh = LogViewerStore.getState().startTime;
+      $scope.initialize();
+      getStatus();
+    }, (err) => {
+      // FIXME: We need to fix this. Right now this fails and we need to handle this more gracefully.
+      $scope.initialize();
+      console.log('ERROR: ', err);
+    });
+  };
+
+  LogViewerStore.subscribe(() => {
+    if(screenSize !== LogViewerStore.getState().fullScreen){
```

Comment:
Seems like this is declared at the top of the file. Any specific reason why this is not attached to `this`. It took a while to see where this was declared.